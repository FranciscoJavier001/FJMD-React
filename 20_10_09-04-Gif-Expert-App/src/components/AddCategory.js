import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => { //** Agregamos el AddCategory pero va directamente al setCategories y va a volver a renderizar, y se recibe props */

    const [inputValue, setInputValue] = useState('') /** Estas son las dos funciones que tiene el inputvalue (donde escribimos) y abajo ya quedo declarado  */

    const handleInputChange = (e) => { 
        setInputValue(e.target.value) /** Esto me permite cambiar el texto */
    }

    const handleSubmit = (e) => { /** Aqui hacemos la peticion para que salga en el DOM, y que se vea reflejado cuando damos enter  */
        e.preventDefault(); /** Esto evita el refresh de la pagina */

        if(inputValue.trim().length > 0){ //** Solo si el arreglo es mayor de 2 que empieze a buscar */
            //** Asi le hizimos para que la primera saliera arriba */
            setCategories(cats => [inputValue, ...cats,]); //** El primer callback es el estado anterior y luego regresa un nuevo estado con todos los nuevos arreglos, se llama a esta instruccion para tener acceso al arreglo, porque se esta pasando como argumento */
            setInputValue('');
        }
    }

    return ( /** Aqui directamente ya estamos en el DOM */
        // El Form agrupa todo lo demas, por eso no hace falta el fragment
        //** Aqui le decimo que va a responder a handleSubmit, osea cuando de enter */
        <form onSubmit={handleSubmit}>  
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange} /** Aqui le doy el valor para cambiarlo */
                />
            </form>
    )
}

AddCategory.prototype = { /** Para que utilicen nuestro componente con el peque√±o candado con los argumentos que estoy esperando, esta es la peticion de los proptypes  */
    setCategories: PropTypes.func.isRequired
}