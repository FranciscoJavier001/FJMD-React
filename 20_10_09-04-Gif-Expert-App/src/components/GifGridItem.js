import React from 'react'

export const GifGridItem = ({title, url}) => { //** Estos son los parametros que se reciben en el map del GifGrid.js */

    return ( //** Es lo que retorna el arreglo de GifGridItem */
        // Estos son los estilos de bootstrap, y aqui vemos que solo se hace el llamado de los parametros dichos en el getGifs.js
        // Como lo vez aqui el div con el className se nos puso a modo de bootstrap, le ponemos el nombre, que en este caso es card y luego la animamos como queramos
        <div className="card animate__animated animate__fadeIn"> 
            {/* Estos son los parametros que trae cada imagen */}
            <img src={url} alt={title}/> 
            {/* Es el titulo que va abajo */}
            <p>{title}</p>
        </div>
    )
}
