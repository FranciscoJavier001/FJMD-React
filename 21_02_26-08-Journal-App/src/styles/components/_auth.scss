
// Esta clase es la que creamos en el div del AuthRouter.js
.auth__main {
    align-items: center; //Quiero centrar todos los items
    background-color: $primary; //El background ya lo tenemos, porque en el styles.scss ya teniamos importado este archivo que nos ayuda a hacer lo que necesitamos
    display: flex; //Es para tener una caja centrada, osea esto permite ponerlo donde quiera
    justify-content: center; //Esto la pone en el centro, cuando estamos viendo el 100% de la pagina
    margin: 0px; //Margin es todo el contexto de la pagina
    height: 100vh; //Esto es la altura y esta del 100 de la altura
    width: 100vw;
}

.auth__box-container {
    background-color: white; //Esto es la caja del Login, osea donde esta escrito
    box-shadow: 0px 3px $dark-grey; //El primero es el sombreado que se va hacia el lado derecho, el segundo es lo que va hacia abajo
    border-radius: 2px; //Para hacer un peque√±o borde redondeado en las esquinas
    padding: 20px; //Esto es lo que se expande laa caja hacia afuera, es de color blanco
    width: 250px; //Esto es el ancho de la caja blanca
}

.auth__title {
    color: darken($color: $primary, $amount: 20); //** Cambio el color del Login */
}