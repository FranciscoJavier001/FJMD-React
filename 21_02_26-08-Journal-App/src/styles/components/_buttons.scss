
.btn { //** Todos los botones van a heredar esta clase */
    background-color: transparent; //** El Boton lo hicimos transparente */
    border: none; //** Le quitamos los borders */
    color: white; //** Las letras las transparentamos con el fondo blanco */
    cursor: pointer; //** Cuando le pasamos el mouse solo sale la manita */
    font-size: 12px; //** Le cambiamos el tamaño a la letra */
    padding: 7px 10px 7px 10px; //** El Primero es hacia arriba, el segundo es (no se), el tercero es hacia abajo, el cuarto es hacia la derecha */

    transition: color .3s ease;

    &:focus { //** Que cuando tenga el focus no salga como un foco verde */
        outline: none; //** Si le doy click no aparece el borde espantoso */
    }

    &:hover {
        color: darken($color: white, $amount: 10);
    }
}

.btn-primary {
    background-color: $primary; //** Ya le dimos color al boton */
    border-radius: 2px; //** Redondiamos los borders */
    
    &:disabled { //** Esto se va a agregar cuando el boton lo desabilitemos */
        background-color: lighten($color: ($primary), $amount: 15); //** Ya tiene una pìnta de boton desabilitado */
    }
    &:hover { //** Cuando el cursor este sobre el */
        background-color: darken($color: $primary, $amount: 15); //** Que se ponga mas oscuro el background del color */
    }
}

.btn-danger{
    background-color: red;

    transition: background-color .3s ease;

    &:hover {
        background-color: darken($color: red, $amount: 10);
    }
}

.btn-block { //** Este es para hacer el boton del tamaño de la caja */
    width: 100%;
}

.google-btn {
    cursor: pointer;
    margin-top: 5px;
    width: 100%;
    height: 42px;
    background-color: $google-blue;
    border-radius: 2px;
    box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);

    transition: box-shadow .3s ease;

    .google-icon-wrapper {
        position: absolute;
        margin-top: 1px;
        margin-left: 1px;
        width: 40px;
        height: 40px;
        border-radius: 2px;
        background-color: $white;
    }
    .google-icon {
        position: absolute;
        margin-top: 11px;
        margin-left: 11px;
        width: 18px;
        height: 18px;
    }
    .btn-text {
        float: right;
        margin: 11px 40px 0 0;
        color: $white;
        font-size: 14px;
        letter-spacing: 0.2px;
    }
    &:hover {
        box-shadow: 0 0 6px $google-blue;
    }
    &:active {
        background: $button-active-blue;
    }
}