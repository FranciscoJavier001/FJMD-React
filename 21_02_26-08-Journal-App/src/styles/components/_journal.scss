
.journal__main-content {
    display: flex;
}

.journal__sidebar { //** Esto es la barra lateral hacia la izquierda */
    background-color: $dark-grey; //** Para que el fondo quede oscuro */
    color: white; //** Para que la letra quede blanca */
    display: flex;
    flex-direction: column;
    height: 100%;
    padding-left: 10px;
    padding-right: 10px;
    width: 450px;
}

.journal__sidebar-navbar {
    display: flex; //** Con Display Flex tenemos la libertad de poner las cosas donde deban de estar */
    justify-content: space-around; //** Con esto se queda en los bordes */
}

.journal__sidebar-navbar h3 {
    font-weight: lighter; //** ESto es el nombre, asi lo hicimos mas delgado */
}

.journal__new-entry { //** Esto es lo que dice New Entry */
    align-items: center; //** Asi alineamos la new entry al centro, de arriba hacia abajo */
    cursor: pointer; //** Para que salga la manita */
    display: flex; //** Para que todo se peda mover */
    flex-direction: column; //** Para que cada cosa este una abajo de la otra */
    justify-content: center; //** Para que todo se quede en el centro */
    margin-top: 30px; //** Asi lo hicimos mas para abajo */
    width: 100%; //** Para que trabaje en toda la anchura */

    transition: color .3s ease;

    &:hover {
        color: darken($color: white, $amount: 20); //** Cambia el icono a un gris */
    }
}