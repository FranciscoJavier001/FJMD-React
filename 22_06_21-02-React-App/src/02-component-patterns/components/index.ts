//** Para no tener un monton de importaciones */

export { ProductButtons } from './ProductButtons';
export { ProductCard } from './ProductCard';
export { ProductImage } from './ProductImage';
export { ProductTitle } from './ProductTitle';