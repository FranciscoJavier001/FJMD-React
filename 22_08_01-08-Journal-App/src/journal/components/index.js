export * from './Navbar';
export * from './SideBar';
export * from './ImageGallery';