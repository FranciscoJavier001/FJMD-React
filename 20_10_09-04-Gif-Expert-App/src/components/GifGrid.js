import React  from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => { //** Aqui lo exportamos para que lo pueda utilizar GifExpertApp, el argumento es category */

    const {data:images, loading} = useFetchGifs(category); //** Es la peticion http para traer las imagenes que corresponden a la categoria, con los : se renombra en la desestructuracion */

    return (
        // Cuando haya que regresar 2 objetos usa el fragment
        <>
            {/* Este h3 son donde se quedan guardadanas nuestras busquedas */}
            <h3 className="card animate__animated animate__fadeIn">{category}</h3>

            {/* Esta es cuando se estan cargando */}
            {loading && <p className="card animate__animated flash">Loading</p>}

            <div className="card-grid">
                {
                    images.map(img => ( //** Asi hago un return de objetos */
                        <GifGridItem //** Aqui se va a mostrar */
                            key={img.id} //** Este es el return ue se me hizo, de id y de img */
                            {...img} //** Operador spred de img  */
                        />
                    ))
                }
            </div>
        </>
    )
}
