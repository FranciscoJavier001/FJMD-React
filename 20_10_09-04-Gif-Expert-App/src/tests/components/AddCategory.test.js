import React from 'react';
import '@testing-library/jest-dom'

import {shallow} from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn() //** Se crea una funcion que no haga nada, pero de esta manera me dice si fue llamada, cuentas veces fue llamada, etc gracias a jest */
    let wrapper = shallow (<AddCategory setCategories={setCategories}/>) //** Aqui se la pasamos como argumento para que pase la prueba, lo subimos porque no lo vamos a cambiar y lo vamos a seguir usando */

    beforeEach(()=> { //** Ciclo de vida de las pruebas, recibe un callback qe es una funcion de flecha, para reinicializarlo y que se limpie */
        jest.clearAllMocks(); //** Se elimina la simulacion de cualquier cosa */
        wrapper = shallow (<AddCategory setCategories={setCategories}/>) //** Reseteamos el componente como estaba originalmente */
    }) 

    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input') //** Referencia al input */
        const value = 'Hola Mundo'

        input.simulate('change', {target: {value: value} }) /**El evento de un handleInputChange se llama de la manera despues de la coma del change, lo que esta adentro de los parentesis/corchetes, que tiene un objeto llamado tarjet y adentro un value */

        expect(wrapper.find('p').text().trim()).toBe(value) //Se busca un parrafo, con el text vemos su contenido, tiene que tener el value que se mando en el 'change'
    })

    test('No debe de postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}}) //** hacemos referencia wrapper.find, donde buscamos el form, y se va a hacer la simulacion del submit, y estos arguentos van a ser, y se va a mandar la funcion de preventDefault */

        expect(setCategories).not.toHaveBeenCalled() //** Es para ver si no se ha lamado, porque el formulario va vacio */
    })
    
})