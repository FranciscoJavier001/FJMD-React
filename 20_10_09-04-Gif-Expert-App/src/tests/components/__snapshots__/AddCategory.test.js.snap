// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
